src/metrics.py

```diff
@@ -20,17 +20,28 @@ def plot_confusion(expected, predicted, mode = True):
         plt.savefig('plots/diff_confusion'+'.png', format='PNG')
     plt.close()
 
-
-def plot_loss(loss_arr, title, val_loss_array=None, mode = True):
-    plt.plot(loss_arr, label = "Train Loss")
-    if val_loss_array!=None:
-        plt.plot(val_loss_array, label = "Val Loss")
-    plt.xlabel('Iterations')
-    plt.ylabel('Loss')
-    plt.title(title)
-    plt.legend()
+def plot_loss(loss_arr, val_loss_array=None, mode = "dcg"):
+    
+    if mode != "dcg" and mode != "diffusion":
+        logging.error('Check value of the mode for plotting. Loss plots not generated.')
+        return
+    if len(loss_arr) >= 50:
+        loss_arr = loss_arr[30:]
+        val_loss_array = val_loss_array[30:]
```

Comment:
What is the 50 and 30? They seem unclear to me.

Reply:
Arbitrary initial values of loss that are omitted during plotting. The code checks if the input array is of length at least 50, and if so, it omits the first 30 values in the array and plots the rest. This is done to avoid plotting the absurdly high initial loss values.

Reply:
Okay